test/cri-containerd/container_test.go

```diff
@@ -621,3 +622,197 @@ func Test_RunContainer_GMSA_WCOW_Hypervisor(t *testing.T) {
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+	client := newTestRuntimeClient(t)
+	podctx := context.Background()
+
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Namespace: testNamespace,
+			},
+			Annotations: map[string]string{
+				"io.microsoft.virtualmachine.computetopology.numa.default": "true",
+			},
+		},
+		RuntimeHandler: wcowHypervisorRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		PodSandboxId: podID,
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageWindowsNanoserver,
+			},
+			Command: []string{
+				"cmd",
+				"/c",
+				"ping",
+				"-t",
+				"127.0.0.1",
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	containerID := createContainer(t, client, ctx, request)
+	runContainerLifetime(t, client, ctx, containerID)
+}
+
+func Test_RunContainer_NUMA_Nodes_Custom_LCOW(t *testing.T) {
+	requireFeatures(t, featureLCOW)
+	testutilities.RequiresBuild(t, 18943)
+
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+	client := newTestRuntimeClient(t)
+
+	podctx := context.Background()
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Namespace: testNamespace,
+			},
+			Annotations: map[string]string{
+				"io.microsoft.virtualmachine.computetopology.numa.settings": "{\"VirtualNodeCount\": 2,\"Settings\": [{\"VirtualNodeNumber\": 0,\"PhysicalNodeNumber\": 0,\"VirtualSocketNumber\": 0,\"CountOfProcessors\": 1,\"CountOfMemoryBlocks\": 512},{\"VirtualNodeNumber\": 1,\"PhysicalNodeNumber\": 0,\"VirtualSocketNumber\": 1,\"CountOfProcessors\": 1,\"CountOfMemoryBlocks\": 512}]}",
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sandboxRequest.Config,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	containerID := createContainer(t, client, ctx, request)
+	runContainerLifetime(t, client, ctx, containerID)
```

Comment:
Can we run something like `lscpu` and verify the NUMA count inside the container? 
I am not sure if something similar can also be done for wcow.

Reply:
lscpu isn't available in the UVM/alpine trust me I've tried :(